Add volumes and volume mounts to run launcher

gibsondan opened the pull request:
Still adding tests that cover the new fields on run launcher. It seems untenable to add config for every type of volume source here: https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Volume.md though..

Test Plan: Integration

<!--- Hello Dagster contributor! It's great to have you with us! -->
<!-- Make sure to read https://docs.dagster.io/community/contributing -->

## Summary
<!-- Describe your changes here, include the motivation/context, test coverage, -->
<!-- the type of change i.e. breaking change, new feature, or bug fix -->
<!-- and related GitHub issue or screenshots (if applicable). -->




## Test Plan
<!--- Please describe the tests you have added and your testing environment (if applicable). -->




## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask in our Slack. -->

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

johannkm reviewed (dismissed):
Does this interact properly with `additional_k8s_volume_mounts` (specified in pipeline/solid tags)? It seems like even before this change `additional_k8s_volume_mounts` wasn't creating volumes, just volume mounts? It's also unclear what should be launcher config and what goes in the tags

gibsondan commented:
@johannkm here's another take on this that I believe addresses your concerns. If you think this approach seems OK at a high-level I'll add some tests

the downside is it uses a private k8s api method to deserialize the volumes into the right model classes (plus it requires constructing an ApiClient, not sure if that is particularly expensive?) - that appears to be the only way to do this currently as per https://github.com/kubernetes-client/python/issues/977

rexledesma commented on python_modules/libraries/dagster-k8s/dagster_k8s/job.py:
```diff
@@ -346,6 +354,37 @@ def config_type_pipeline_run(cls, default_image_pull_policy=None):
+                    Shape(
+                        {
+                            "name": str,
+                            "path": str,
```
```suggestion
                            "mount_path": str,
```

Also, what about `mount_propogation` and `sub_path_expr`?

gibsondan replied:
I think when in doubt we should match the k8s API rather than innovating

gibsondan replied:
oh i misunderstood, you meant we should fill out the whole VolumeMount API. Good point

gibsondan commented:
Current dependencies on/for this PR:
* master
  * **PR #4872**  (this PR)
    * **PR #4947** 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh) and will continue to be automatically updated while this PR remains open.

johannkm reviewed (approved):
I'm happy with this, thanks for pushing it through.
